File: F#/day-1.fs
module DayOne

open System.IO

let tryParseInt s =
    try
        s |> int |> Some
    with _ ->
        None

let partOneSolver inputPath =
    File.ReadAllLines inputPath
    |> Array.fold (fun stateList nextLine ->
        match tryParseInt nextLine with
        | Some x ->
            (x :: (stateList |> List.item 0)) :: (stateList |> List.removeAt 0)
        | None -> [] :: stateList) [[]]
    |> List.map List.sum
    |> List.max

let partTwoSolver inputPath =
    File.ReadAllLines inputPath
    |> Array.fold (fun stateList nextLine ->
        match tryParseInt nextLine with
        | Some x ->
            (x :: (stateList |> List.item 0)) :: (stateList |> List.removeAt 0)
        | None -> [] :: stateList) [[]]
    |> List.map List.sum
    |> List.sortDescending
    |> List.take 3
    |> List.sum

File: C#/day-1/day-1.cs
using System.Linq;

public static class DayOne {
    private static int? ParseIntNullable(string s) {
        if (int.TryParse(s, out int x)) {
            return x;
        }
        return null;
    }

    public static int PartOneSolver(string inputPath) =>
        File.ReadAllLines(inputPath)
            .Select(ParseIntNullable)
            .Aggregate(new List<List<int>>() {new List<int>()}, (state, x) => {
                if (x is null) {
                    return state.Append(new List<int>()).ToList();
                }
                else {
                    state.Last().Add((int)x);
                    return state;
                }
            })
            .Select(x => x.Sum())
            .Max();

    public static int PartTwoSolver(string inputPath) =>
        File.ReadAllLines(inputPath)
            .Select(ParseIntNullable)
            .Aggregate(new List<List<int>>() {new List<int>()}, (state, x) => {
                if (x is null) {
                    return state.Append(new List<int>()).ToList();
                }
                else {
                    state.Last().Add((int)x);
                    return state;
                }
            })
            .Select(x => x.Sum())
            .OrderByDescending(x => x)
            .Take(3)
            .Sum();
}